internal/onprem/event_extractor.go

```diff
@@ -69,7 +69,7 @@ func (e *EventExtractor) ExtractEvents(tarFilename string) (chan types.EventEnve
 	e.logger.WithFields(logrus.Fields{
 		"directory": tmpdir,
 	}).Info("extracting events from files in directory")
-	e.extractEvents(tmpdir, eventChannel)
+	e.extractEvents(tmpdir, tarFilename, eventChannel)
```

Comment:
What about passing the logger around (or contained in a context) and keep adding fields in it?